src/pybind/mgr/dashboard/controllers/smb.py

```diff
@@ -0,0 +1,188 @@
+            "source_type": (str, "resource"),
+            "ref": (str, "Reference identifier for the join auth resource")
+        }], "List of join auth sources for domain settings")
+    }, "Domain-specific settings for active-directory auth mode"),
+    "user_group_settings": ([{
+        "source_type": (str, "resource"),
+        "ref": (str, "Reference identifier for the user group resource")
+    }], "User group settings for user auth mode"),
+    "custom_dns": ([str], "List of custom DNS server addresses"),
+    "placement": ({
+        "count": (int, "Number of instances to place")
+    }, "Placement configuration for the resource")
+}
+
+CLUSTER_SCHEMA_RESULTS = {
+    "results": ([{
+        "resource": ({
+            "resource_type": (str, "ceph.smb.cluster"),
+            "cluster_id": (str, "Unique identifier for the cluster"),
+            "auth_mode": (str, "Either 'active-directory' or 'user'"),
+            "intent": (str, "Desired state of the resource, e.g., 'present' or 'removed'"),
+            "domain_settings": ({
+                "realm": (str, "Domain realm, e.g., 'DOMAIN1.SINK.TEST'"),
+                "join_sources": ([{
+                    "source_type": (str, "resource"),
+                    "ref": (str, "Reference identifier for the join auth resource")
+                }], "List of join auth sources for domain settings")
+            }, "Domain-specific settings for active-directory auth mode"),
+            "user_group_settings": ([{
+                "source_type": (str, "resource"),
+                "ref": (str, "Reference identifier for the user group resource")
+            }], "User group settings for user auth mode (optional)"),
+            "custom_dns": ([str], "List of custom DNS server addresses (optional)"),
+            "placement": ({
+                "count": (int, "Number of instances to place")
+            }, "Placement configuration for the resource (optional)"),
+        }, "Resource details"),
+        "state": (str, "State of the resource"),
+        "success": (bool, "Indicates whether the operation was successful")
+    }], "List of results with resource details"),
+    "success": (bool, "Overall success status of the operation")
+}
+
+LIST_CLUSTER_SCHEMA = [CLUSTER_SCHEMA]
+
+SHARE_SCHEMA = {
+    "resource_type": (str, "ceph.smb.share"),
+    "cluster_id": (str, "Unique identifier for the cluster"),
+    "share_id": (str, "Unique identifier for the share"),
+    "intent": (str, "Desired state of the resource, e.g., 'present' or 'removed'"),
+    "name": (str, "Name of the share"),
+    "readonly": (bool, "Indicates if the share is read-only"),
+    "browseable": (bool, "Indicates if the share is browseable"),
+    "cephfs": ({
+        "volume": (str, "Name of the CephFS file system"),
+        "path": (str, "Path within the CephFS file system"),
+        "provider": (str, "Provider of the CephFS share, e.g., 'samba-vfs'")
+    }, "Configuration for the CephFS share")
+}
```

Comment:
Oh okay, that makes sense, did not like schemas that much but, I'm seeing one big difference with the models, the schema provides field explanation which the models do not, also schemas show up on the swagger where as models do not:

[screen-capture (13).webm](https://github.com/user-attachments/assets/262ae294-2623-40e1-9d48-79f6c2e23995)

I personally find it helpful having the return objects and field descriptions in the swagger

Reply:
oh yeah, good catch. hmm I think that's something we need to improve on our RESTController then!

Reply:
Okay, then for now I'll leave it as it is, and will look into this for following PRs.

Ideally we move this schemas to their own `models/smb.py` and have them as `NamedTuple` instead of python objects

Reply:
sounds good, thanks